sound/soc/sof/intel/hda-dai.c

```diff
@@ -134,21 +204,32 @@ static int hda_link_hw_params(struct snd_pcm_substream *substream,
 	struct sof_intel_hda_stream *hda_stream;
 	struct hda_pipe_params p_params = {0};
 	struct hdac_ext_link *link;
-	int stream_tag;
+	int stream_tag, ret;
 
-	link_dev = snd_soc_dai_get_dma_data(dai, substream);
+	link_dev = hda_link_stream_assign(bus, substream);
+	if (!link_dev)
+		return -EBUSY;
```

Comment:
why -EBUSY?

Reply:
@plbossart -EBUSY because hda_link_stream_assign returns a NULL when there are no free link DMA channels available. Is that acceptable?